src/ai/backend/manager/idle.py

```diff
@@ -706,22 +751,50 @@ def _avg(util_list: list[float]) -> float:
             self._redis_live,
             lambda r: r.set(
                 util_last_collected_key,
-                f"{t:.06f}",
+                f"{util_now:.06f}",
                 ex=max(86400, int(self.time_window.total_seconds() * 2)),
             ),
         )
 
-        if not_enough_data:
-            await self._report_redis(redis_obj, session_id, avg_utils)
-            return True
+        def _avg(util_list: list[float]) -> float:
+            try:
+                return sum(util_list) / len(util_list)
+            except ZeroDivisionError:
+                return 0.0
 
-        # Check over-utilized (not to be collected) resources.
-        sufficiently_utilized = {
-            k: (float(avg_utils[k]) >= float(threshold))
+        avg_utils: Mapping[str, float] = {k: _avg(v) for k, v in util_series.items()}
+
+        # util_avg_thresholds's key is resource name
+        # and value is `(avg_util, threshold)`
+        util_avg_thresholds: dict[str, UtilizationExtraInfo] = {
+            k: UtilizationExtraInfo(float(avg_utils[k]), float(threshold))
             for k, threshold in self.resource_thresholds.items()
             if (threshold is not None) and (k not in unavailable_resources)
         }
+        for k in util_avg_thresholds:
+            if k in ("mem", "cuda_mem"):
```

Comment:
Is there a way to generalize detection of memory resources, so that we can support all other accelerators (TPU, ROCm, ...) also on idle checker?

Reply:
I still have no idea.. I have been thinking of applying such unit type.